test/broadcast.jl

```diff
@@ -300,7 +300,6 @@ import Base.Meta: isexpr
 # PR 16988
 @test Base.promote_op(+, Bool) === Int
 @test isa(broadcast(+, [true]), Array{Int,1})
-@test Base.promote_op(Float64, Bool) === Float64
```

Comment:
I returns `Any` because I remove the specialized method when the first argument is a type. `promote_op` would only be used by the unary and binary elementwise operations in base, as well as by the matrix multiplication methods, where we know that the operator is not a type.

I guess I could leave the definition for uses outside Base addressing @martinholters concern above.